### ISSUE TITLE
WPS Pin attack not giving IP to the victim

### ISSUE THREAD

**Issue by Mrtlsn64r02a662n** [Koala633/hostbase#4]
Hi Koala, i'm running WPS Pin attack (Rogue AP with hostapd and the wps push button) . 
After everything is started (Hostapd, etc) i use my phone to connect to the FakeAp, but when i connect to it i do not receive an IP. 

I'm running hostbase1.0 onto a fresh Kali 2017.1 installed on the hard disk. 

newinstall.sh was executed correctly. 

The adapter is a TP-LINK TL-WN722N.

**Comment by Mrtlsn64r02a662n** [comment]
Ok, i figured why It does not receive an IP: 
because in hostapd.conf the fake ap is configured as a wpa network with password  12345678. 
How Can a victim connect to a wpa protected Ap without knowing the password? 
Would not It Be better to create an open fake ap?

**Comment by kcdtv** [comment]
> How Can a victim connect to a wpa protected Ap without knowing the password?

Using the WPS PBC protocol, that's the point of this attack, get the victim on a rogue **WPA** access point (and yes, without him knowing the key... magical) 

> Would not It Be better to create an open fake ap?

It is a different attack.
That's a standard rogue AP on **open** network with  captive portal
That can be done too with hostbase but it is nothing new.
The **WPS PBC** rogue approach is one of the new feature that offers hostbase,

**Comment by Mrtlsn64r02a662n** [comment]
_Using the WPS PBC protocol, that's the point of this attack, get the victim on a rogue WPA access point (and yes, without him knowing the key... magical)_

OK, without entering into details of this magical attack, i have to say that it does not work, at least in my experiment.
I run this attack from my Kali PC running hostbase BUT when i try to connect to rogue ap from my Android phone i was unable to connect to rogue ap, unless i insert the wpa key 12345678: only in this case i was able to connect to rogue ap, obtain an ip and get the fake web page telling to push wps button. 

Basically, from my android phone i see the rogue ap as a WPA protected network, so i can't connect to it unless i write the wpa key 12345678.

This is the msgs appearing in hostbase window when i write in a random password (not 12345678)

wlan1: AP-ENABLED
wlan1: WPS-PBC-ACTIVE
wlan1, STA 00:08:22;ea:86:6a authenticated
wlan1, STA 00:08:22;ea:86:6a  associated (aid 1)
WPA: wpa_sm_step() called recursively
wlan1, STA 00:08:22;ea:86:6a  deauthenticated due to local deauth request
wlan1: WPS TIMEOUT

**Comment by kcdtv** [comment]
you should givre really much much more details...
 It looks like your card might have some problems with hostapd checking the error mesage. 
what is the chipset of your wifi interface?

**Comment by Mrtlsn64r02a662n** [comment]
Well, usually my card works fine with hostapd, except with this Hostbase Wps Pbc attack.

The chipset is **Atheros Communications, Inc. AR9271 802.11n**
The driver is **ath9k_htc**

As i said, when i try to connect to rogue ap from my android phone i see it as a Wpa-protected network:  
1) is it fine or should i see it as a open network? 
2) assuming it is fine: which wpa key should i use to connect?

**Comment by Koala633** [comment]
Hi, just to be sure, did you start the wps_pbc on your android ? You cant connect an android like à computer just by clicking on the network.You have to go  on the setting and push something like "wps_connect".In another case it will always ask you the key.

**Comment by Mrtlsn64r02a662n** [comment]
Hi, no: i did not know anything about this wps_connect on android. But i
suppose that also a real Android "victim" would be in the same situation:
he is asked for a Wpa Key.

Anyway, i made another test from a Windows "victim": trying to connect to
hostbase rogue ap the victim *is asked a WPS PIN*. Which Pin should he/she
insert in order to connect to rogue ap?

2017-06-13 9:14 GMT+02:00 Koala633 <notifications@github.com>:

**Comment by Koala633** [comment]
When you see the Windows with the network around, the wps setting is on the top right corner(you also have the wifi direct option on the same tab)

The problem with Android is if the victim doesn't know how to connect with his phone using wps it will stuck at the key.For à computer its more easy.In the other case Windows do not have to ask the pin.I make my test on win7 and win10  full updated, which version off Windows you use ? also pbc_in_m1 is set on every hostapd.conf file, There is no pin configured at all, let me know which version of Windows you use and which attack with the script.

**Comment by Mrtlsn64r02a662n** [comment]
Ok, let's concentrate on Windows victims...

I'm running WPS Pin attack (Rogue AP with hostapd and the wps push button , the "simple" one, non multi-AP). 

I'm using Windows 7 Professional: when i try to connect to RogueAP i'm asked for a Wps Pin.

**Comment by Koala633** [comment]
Very strange, i suppose this is à driver wich not reconize the pbc method, did you try with a external wireless adapter ?

**Comment by Mrtlsn64r02a662n** [comment]
It's CENTRINO INTEL.
No, i do not have another external adapter besides the TP-LINK which i use
to create the Rogue AP.


2017-06-13 17:33 GMT+02:00 Koala633 <notifications@github.com>:

**Comment by kcdtv** [comment]
Do that:
 - Erase your connection profile to your own PA
 - Enable WPS on your PA
 - Connect to your PA with your Intel card 
And check if the windows network manager propose you or not to push the PBC button form your access point.

**Comment by Mrtlsn64r02a662n** [comment]
Sorry but i can't do that: my Intel machine is at office and i am not allowed to take it home. 
So this way is not viable.

Instead, i am giving a look at another hostbase attack: 
**1.  Rogue AP with airbase**

Would you please briefly describe how does this work?

**Comment by kcdtv** [comment]
This is git hub, not a support forum, stay focus :smiley_cat:  
it is quite clear that you did not read enough the PDF: The PBC attack is only for windows unless you voluntary use PBC from android 
As you cannot perform the test there is no much to stay: your intel card or windows version does not have wps PBC support. It seems logical in a working environment 
issue is closed.
